WordPress/src/main/java/org/wordpress/android/ui/mysite/ScanAndBackupSource.kt

```diff
@@ -0,0 +1,34 @@
+package org.wordpress.android.ui.mysite
+
+import kotlinx.coroutines.flow.flow
+import org.wordpress.android.ui.jetpack.JetpackCapabilitiesUseCase
+import org.wordpress.android.ui.mysite.MySiteUiState.PartialState.JetpackCapabilities
+import org.wordpress.android.util.config.BackupScreenFeatureConfig
+import org.wordpress.android.util.config.ScanScreenFeatureConfig
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ScanAndBackupSource
+@Inject constructor(
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val scanScreenFeatureConfig: ScanScreenFeatureConfig,
+    private val backupScreenFeatureConfig: BackupScreenFeatureConfig,
+    private val jetpackCapabilitiesUseCase: JetpackCapabilitiesUseCase
+) : MySiteSource<JetpackCapabilities> {
+    override fun buildSource(siteId: Int) = flow {
+        emit(JetpackCapabilities(scanAvailable = false, backupAvailable = false))
+        val site = selectedSiteRepository.getSelectedSite()
+        if (site == null || site.id != siteId) return@flow
+        if (scanScreenFeatureConfig.isEnabled() || backupScreenFeatureConfig.isEnabled()) {
+            val itemsVisibility = jetpackCapabilitiesUseCase.getJetpackPurchasedProducts(site.siteId)
+            emit(
+                    JetpackCapabilities(
+                            scanAvailable = scanScreenFeatureConfig.isEnabled() && itemsVisibility.scan,
+                            backupAvailable = backupScreenFeatureConfig.isEnabled() && itemsVisibility.backup
+                    )
+            )
+        }
+    }
+}
```

Comment:
This is really neat! The only thing I'm not sure if I understand is why we still need to compare the `siteId` argument with the id from `selectedSiteRepository.getSelectedSite()`. I'm not very familiar with the Flow API, but since we're calling `buildSource` and creating a new Flow every time the site is switched, I'd expect the previous Flow to be cancelled, since we're not "subscribed" to it anymore. Is that not the case?